### DISCUSSION TITLE
Zone cleaning and Alexa

### DISCUSSION THREAD

**Discussion by Raphael909** [PiotrMachowski/lovelace-xiaomi-vacuum-map-card#352] [Q&A] [answered]
First, thank you for sharing this wonderful integration!

 I searched in both card and extractor project but could not find any discussion or guidance around exposing zone cleaning to voice assistant. Either for Alexa or Google assistant. 
Is there a way to achieve something like "Alexa clean zone x " ?
Any thoughts? 

Really appreciate your effort and time!

**Comment by PiotrMachowski** [comment, marked as answer]
You probably have to make a script for every room. It's not possible via map card and map extractor

**Comment by dominikreinelt** [comment]
@PiotrMachowski Thanks for the really good solutions.
I still have one question, how can I create the script so that I can output a zone as a switch in Alexa or HomeKit? 
I hope you can help me. 
Best regards Dominik

**Reply by PiotrMachowski** [reply]
Open card editor, select a zone on a map and click "copy service call" in the editor

**Reply by dominikreinelt** [reply]
okay yes that's no problem, and then I have to store that via the file editor under /config/scripts.yaml, is that correct? How do I then get that as a device or entity so that I can then Expose it via the HomeKit integration? Sorry for the stupid questions :)

**Reply by PiotrMachowski** [reply]
You have to adjust copied json to make it compatible with the target formatting and use it in [template switch](https://www.home-assistant.io/integrations/switch.template/)
